tool/src/c/formatter.rs

```diff
@@ -28,14 +28,27 @@ impl<'tcx> CFormatter<'tcx> {
     }
 
     /// Resolve and format a named type for use in code (without the namespace)
-    pub fn fmt_type_name(&self, id: TypeId) -> Cow<'tcx, str> {
-        let resolved = self.tcx.resolve_type(id);
-        let name = resolved.name().as_str().into();
+    pub fn fmt_type_name(&self, id: SymbolId) -> Cow<'tcx, str> {
```

Comment:
issue: i think this should still continue to work on TypeId, with a separate one for TraitId.

Overall I don't think it's super valuable to treat types/traits as two variants of the same thing

Reply:
i think the main place you want to use SymbolId is in things like fmt_decl_header_path